### ISSUE TITLE
Relax rules for trailing whitespaces in format specifier

### ISSUE THREAD

**Issue by liamlim** [dotnet/runtime#35615]
In short please allow to write this:

```csharp
var a = new DateTime();
var b = $"Value: '{ a:yyyy-MM-dd }'"; // trailing space in interpolated string => illegal
```

Currently it is legal code to write this:
```csharp
var a = new DateTime();
var b = $"Value '{ a:yyyy-MM-dd}'"; // no trailing space in interpolated string => legal
```

The current legal code looks a bit ugly because I prefer to add spaces inside interpolated strings. Also it's inconsistent with my code because I use spaces inside interpolated strings in all places except when I want to use format specifier which is also the reason I don't use this feature and usually use some workarounds.

**Comment by tannergooding** [comment]
@gafter, I don't believe this is a runtime issue. The error reported is `error CS8088: A format specifier may not contain trailing whitespace.` and it is completely legal code according to the framework, so I believe this is purely a language limitation.

```
C:\Program Files (x86)\Microsoft Visual Studio\2019\Community>csi
Microsoft (R) Visual C# Interactive Compiler version 3.5.0-beta4-20153-05 ()
Copyright (C) Microsoft Corporation. All rights reserved.

Type "#help" for more information.
> var a = new DateTime();
> a.ToString("yyyy-MM-dd")
"0001-01-01"
> a.ToString(" yyyy-MM-dd ")
" 0001-01-01 "
>
```

**Comment by tannergooding** [comment]
Additionally, it appears as though the compiler already strips out leading whitespace from the resulting IL: https://sharplab.io/#v2:EYLgxg9gTgpgtADwGwBYA+ABATARgLABQGAzAATakDCpA3oaQ+WRiqQLIAUAlLfY/wDcAhlFJDSAXlIA7GAHdSAESEAXGABUAlgFsY3ANx9+DYaOCTSAEgBEANSEAbAK4wQpAOQ0xIAJ5+fcGxscAAmIQC+7tb6pAD0saQqUEKaDprSAOakAM4ADkJgMKTpxdJqULkQDqowIeQ4AAySAHzFDg4wGY5GpOGE4UA==
